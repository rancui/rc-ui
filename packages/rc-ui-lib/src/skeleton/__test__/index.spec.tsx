import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { Skeleton } from '..';

describe('Skeleton', () => {
  afterEach(() => {
    cleanup();
    jest.restoreAllMocks();
  });

  it('should render with row width array correctly', async () => {
    const { container } = render(<Skeleton row={4} rowWidth={['100%', 30, '5rem']} />);
    expect(container).toMatchSnapshot();
  });

  it('should render with row height array correctly', async () => {
    const { container } = render(<Skeleton row={4} rowHeight={['10px', 30, '1rem']} />);
    expect(container).toMatchSnapshot();
  });

  it('should render default slot when loading is false', () => {
    const { container } = render(
      <Skeleton loading={false}>
        <div>Content</div>
      </Skeleton>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should change avatar size when using avatar-size prop', () => {
    const { container } = render(<Skeleton avatar avatarSize="20rem" />);

    const avatar = container.querySelector('.rc-skeleton__avatar');
    expect(getComputedStyle(avatar).width).toMatchSnapshot('20rem');
    expect(getComputedStyle(avatar).height).toMatchSnapshot('20ren');
  });

  it('should change avatar shape when using avatar-shape prop', () => {
    const { container } = render(<Skeleton avatar avatarShape="square" />);
    expect(container.querySelector('.rc-skeleton__avatar')).toMatchSnapshot();
  });

  it('should be round when using round prop', () => {
    const { container } = render(<Skeleton title round avatar />);
    expect(container.querySelector('.rc-skeleton--round')).toBeTruthy();
  });

  it('should allow to disable animation', async () => {
    const { container, rerender } = render(<Skeleton row={1} />);

    expect(container.querySelector('.rc-skeleton--animate')).toBeTruthy();

    rerender(<Skeleton row={1} animate={false} />);
    expect(container.querySelector('.rc-skeleton--animate')).toBeFalsy();
  });
});
