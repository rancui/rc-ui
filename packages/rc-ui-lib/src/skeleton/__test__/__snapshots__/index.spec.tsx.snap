// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton should change avatar shape when using avatar-shape prop 1`] = `
<div
  class="rc-skeleton__avatar rc-skeleton__avatar--square"
/>
`;

exports[`Skeleton should change avatar size when using avatar-size prop: 20rem 1`] = `"20rem"`;

exports[`Skeleton should change avatar size when using avatar-size prop: 20ren 1`] = `"20rem"`;

exports[`Skeleton should render default slot when loading is false 1`] = `
<div>
  <div>
    Content
  </div>
</div>
`;

exports[`Skeleton should render with row height array correctly 1`] = `
<div>
  <div
    class="rc-skeleton rc-skeleton--animate rc-skeleton--round"
  >
    <div
      class="rc-skeleton__content"
    >
      <div
        class="rc-skeleton__row"
        style="width: 100%; height: 10px;"
      />
      <div
        class="rc-skeleton__row"
        style="width: 100%; height: 30px;"
      />
      <div
        class="rc-skeleton__row"
        style="width: 100%; height: 1rem;"
      />
      <div
        class="rc-skeleton__row"
        style="width: 60%;"
      />
    </div>
  </div>
</div>
`;

exports[`Skeleton should render with row width array correctly 1`] = `
<div>
  <div
    class="rc-skeleton rc-skeleton--animate rc-skeleton--round"
  >
    <div
      class="rc-skeleton__content"
    >
      <div
        class="rc-skeleton__row"
        style="width: 100%;"
      />
      <div
        class="rc-skeleton__row"
        style="width: 30px;"
      />
      <div
        class="rc-skeleton__row"
        style="width: 5rem;"
      />
      <div
        class="rc-skeleton__row"
      />
    </div>
  </div>
</div>
`;
